src/main/java/core/mate/academy/service/impl/ExcavatorProducerImpl.java

```diff
@@ -0,0 +1,36 @@
+package core.mate.academy.service.impl;
+
+import core.mate.academy.model.Excavator;
+import core.mate.academy.service.MachineProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducerImpl implements MachineProducer<Excavator> {
+    private final List<Excavator> excavators;
```

Comment:
Access modifier for the 'excavators' field should be specified to ensure encapsulation. Consider using 'private' to restrict direct access to the field.